packages/docsearch-react/src/DocSearchModal.tsx

```diff
@@ -328,6 +329,22 @@ export function DocSearchModal({
     }
   }, [initialQuery, refresh]);
 
+  React.useEffect(() => {
+    const setFullViewportHeight = () => {
+      // We rely on a CSS property to set the modal height to the full viewport height
+      // because all mobile browsers don't compute their height the same way.
+      // See https://css-tricks.com/the-trick-to-viewport-units-on-mobile/
+      const vh = window.innerHeight * 0.01;
```

Comment:
This needs to be computed only if next line applies, so we can wrap all the function body with the `modalRef.current` check.